cgen: method on array sumtype with some already used method names

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
```
OS: macos, macOS, 12.3.1, 21E258
Processor: 8 cpus, 64bit, little endian, Apple M1
CC version: Apple clang version 13.1.6 (clang-1316.0.21.2.3)

getwd: /Users/hunam/Desktop
vmodules: /Users/hunam/.vmodules
vroot: /Applications/v
vexe: /Applications/v/v
vexe mtime: 2022-05-16 13:22:42
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 ac90a2b.e5c7fe3

Git version: git version 2.32.0 (Apple Git-132)
Git vroot status: weekly.2022.19-65-ge5c7fe30
.git/config present: true
thirdparty/tcc status: thirdparty-unknown-unknown de82a130
```

<!-- Please include the standalone minimal reproducible code.
If not, it may be closed by the administrators. -->
**What did you do?**
```v
type Alias = []int

fn (a Alias) last() {}

fn main() {
	Alias([0]).last()
}
```
Note that this also appends with the method named `pop` but not `remove_last` for some reason.

**What did you expect to see?**
Either nothing as this code is dumb or an error saying that we cannot overwrite a method, I don't quite know.
I stand with the first idea because in Go this is allowed.
 
**What did you see instead?**
```c
/tmp/v_501/report.8485403898112802553.tmp.c:2067:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t1.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/report.8485403898112802553.tmp.c:11517:10: error: operand of type 'void' where arithmetic or pointer type is required
        (*(int*)main__Alias_last(((new_array_from_c_array(1, 1, sizeof(int), _MOV((int[1]){0}))))));
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning and 1 error generated.
```

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -941,7 +941,7 @@ fn (mut g Gen) method_call(node ast.CallExpr) {
 	if left_sym.kind == .map && node.name in ['clone', 'move'] {
 		receiver_type_name = 'map'
 	}
-	if final_left_sym.kind == .array
+	if final_left_sym.kind == .array && !(left_sym.kind == .alias && left_sym.has_method(node.name))
 		&& node.name in ['repeat', 'sort_with_compare', 'free', 'push_many', 'trim', 'first', 'last', 'pop', 'clone', 'reverse', 'slice', 'pointers'] {
 		if !(left_sym.info is ast.Alias && typ_sym.has_method(node.name)) {
 			// `array_Xyz_clone` => `array_clone`

--- a/vlib/v/tests/alias_array_has_method_test.v
+++ b/vlib/v/tests/alias_array_has_method_test.v
@@ -1,0 +1,11 @@
+type Alias = []int
+
+fn (a Alias) last() int {
+	return 22
+}
+
+fn test_alias_array_has_method() {
+	ret := Alias([0]).last()
+	println(ret)
+	assert ret == 22
+}

